jaeger/info.go

```diff
@@ -0,0 +1,46 @@
+package jaeger
+
+import (
+	"errors"
+	"net/http"
+	"net/url"
+	"path"
+
+	"github.com/kiali/kiali/config"
+)
+
+func getJaegerInfo(client http.Client, endpoint *url.URL) (*JaegerInfo, int, error) {
+	jaegerConfig := config.Get().ExternalServices.Tracing
+	integration := true
+	error := ""
+	if !jaegerConfig.Enabled {
+		return nil, http.StatusNoContent, nil
+	}
+
+	externalUrl := jaegerConfig.URL
+	if externalUrl == "" {
+		return nil, http.StatusServiceUnavailable, errors.New("Jaeger URL is not set in Kiali configuration")
+	}
+
+	if jaegerConfig.InClusterURL == "" {
+		return nil, http.StatusServiceUnavailable, errors.New("Jaeger URL in cluster is not set in Kiali configuration")
+	}
+
+	u := endpoint
+	u.Path = path.Join(u.Path, "/api/services")
+
+	_, code, err := makeRequest(client, u.String(), nil)
+	if err != nil || code != 200 {
+		integration = false
+		error = "Error with internal connection with Jaeger: " + err.Error()
```

Comment:
you will have an issue here when `err` is `nil` (when `err` is `nil` and `code` is anything but 200, the `if` condition is true)

Reply:
yep, the point is taht jaeger returns 400 if there is any error and we are showing these errors in kliali UI,

for example :
/api/traces?end=1578581310315000&limit=20&lookback=1h&maxDuration=12ms&minDuration=22ms&service=details.bookinfo&start=1578577710315000


return this json with code 400, that means that jaeger is working but the query is wrong
```json
{"data":null,"total":0,"limit":0,"offset":0,"errors":[{"code":400,"msg":"'maxDuration' should be greater than 'minDuration'"}]}
```

Reply:
Yes, but given your `if` condition above, `err` can be nil. So your code will crash (`err.Error()` will crash)

Reply:
o.O you're right OMG